siren/src/main/java/siren/components/Plan.java

```diff
@@ -1,20 +1,69 @@
 package siren.components;
 
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
 import siren.jobs.Employee;
 
 public class Plan {
 
+	private int daysInMonth;
 	private Map<Shift, Set<Employee>> plan;
 
-	public Plan(Map<Shift, Set<Employee>> plan) {
+	public Plan(int daysInMonth, Map<Shift, Set<Employee>> plan) {
 		this.plan = plan;
+		this.daysInMonth = daysInMonth;
 	}
 
 	public Map<Shift, Set<Employee>> getPlan() {
 		return plan;
 	}
 
+	public int getDays() {
+		return daysInMonth;
+	}
+
+	public int sumWorkingHours(Employee employee) {
+		int sum = 0;
+		for (Shift s : plan.keySet()) {
+			if (plan.get(s).contains(employee)) {
+				sum += s.duration();
+			}
+		}
+		return sum;
+	}
+
+	public boolean isNormalShiftDay(Employee employee, int day) {
+		for (Shift shift : filterByDay(day)) {
+			// TODO only hits the first shift in a day, ordering not ensured
+			return shift.getCarType() != null;
+		}
+		// may throw an exception
+		return false;
+	}
+
+	public int getTime(Employee employee, int actDay, boolean start) {
```

Comment:
The name of the function involve the type of the time it gets.

Reply:
Thanks, The function and parameter names are misleading... I will correct that in a next branch
In this case the getTime name means a Hour on a day.

I think the "actDay" parameter name should be corrected to "day" and "start" parameter to -> I have no simple name actually, Do you suggest something? Or shall be refactor as an Enum with EShift.START and EShift.END instead of true or false.

I will do that soon on a Correction branch. Thank you!